backend/lucene/src/main/java/org/hibernate/search/backend/lucene/search/impl/LuceneSearchScopeModel.java

```diff
@@ -76,49 +76,51 @@ public EventContext getIndexesEventContext() {
 			IndexSchemaFieldNodeComponentRetrievalStrategy<T> componentRetrievalStrategy) {
 		LuceneIndexModel indexModelForSelectedSchemaNode = null;
 		LuceneIndexSchemaFieldNode<?> selectedSchemaNode = null;
-		T selectedComponent = null;
-		LuceneCompatibilityChecker converterChecker = null;
+		LuceneScopedIndexFieldComponent<T> scopedIndexFieldComponent = new LuceneScopedIndexFieldComponent<>();
 
 		for ( LuceneIndexModel indexModel : indexModels ) {
 			LuceneIndexSchemaFieldNode<?> schemaNode = indexModel.getFieldNode( absoluteFieldPath );
+			if ( schemaNode == null ) {
+				continue;
+			}
 
-			if ( schemaNode != null ) {
-				T component = componentRetrievalStrategy.extractComponent( schemaNode );
+			T component = componentRetrievalStrategy.extractComponent( schemaNode );
+			if ( selectedSchemaNode == null ) {
+				selectedSchemaNode = schemaNode;
+				indexModelForSelectedSchemaNode = indexModel;
+				scopedIndexFieldComponent.setComponent( component );
+				continue;
+			}
 
-				if ( selectedSchemaNode == null ) {
-					selectedSchemaNode = schemaNode;
-					indexModelForSelectedSchemaNode = indexModel;
-					selectedComponent = component;
-				}
-				else if ( !componentRetrievalStrategy.hasCompatibleCodec( selectedComponent, component ) ) {
-					throw componentRetrievalStrategy.createCompatibilityException(
-							absoluteFieldPath,
-							selectedComponent,
-							component,
-							EventContexts.fromIndexNames(
-									indexModelForSelectedSchemaNode.getIndexName(),
-									indexModel.getIndexName()
-							)
-					);
-				}
-				else if ( !componentRetrievalStrategy.hasCompatibleConverter( selectedComponent, component ) ) {
-					converterChecker = new LuceneFailingCompatibilityChecker(
-							absoluteFieldPath, selectedComponent, component, EventContexts.fromIndexNames(
-							indexModelForSelectedSchemaNode.getIndexName(),
-							indexModel.getIndexName()
-					), componentRetrievalStrategy );
-				}
+			if ( !componentRetrievalStrategy.hasCompatibleCodec( scopedIndexFieldComponent.getComponent(), component ) ) {
+				throw componentRetrievalStrategy.createCompatibilityException(
+						absoluteFieldPath,
+						scopedIndexFieldComponent.getComponent(),
+						component,
+						EventContexts.fromIndexNames(
+								indexModelForSelectedSchemaNode.getIndexName(),
+								indexModel.getIndexName()
+						)
+				);
+			}
+
+			LuceneFailingCompatibilityChecker failingCompatibilityChecker = new LuceneFailingCompatibilityChecker(
+					absoluteFieldPath, scopedIndexFieldComponent.getComponent(), component, EventContexts.fromIndexNames(
+					indexModelForSelectedSchemaNode.getIndexName(),
+					indexModel.getIndexName()
+			), componentRetrievalStrategy );
```

Comment:
This is indented quite strangely. Even automatic formatting in IDEA (which usually isn't great) looks better.